component/vdr/cheqd/vdr.go

```diff
@@ -0,0 +1,245 @@
+	PublicKey            []map[string]interface{} `json:"publicKey,omitempty"`
+	Service              []map[string]interface{} `json:"service,omitempty"`
+	Authentication       []interface{}            `json:"authentication,omitempty"`
+	AssertionMethod      []interface{}            `json:"assertionMethod,omitempty"`
+	CapabilityDelegation []interface{}            `json:"capabilityDelegation,omitempty"`
+	CapabilityInvocation []interface{}            `json:"capabilityInvocation,omitempty"`
+	KeyAgreement         []interface{}            `json:"keyAgreement,omitempty"`
+	Created              *time.Time               `json:"created,omitempty"`
+	Updated              *time.Time               `json:"updated,omitempty"`
+	Proof                []interface{}            `json:"proof,omitempty"`
+}
+
+// RawDocCheqd type.
+type RawDocCheqd struct {
+	DidDocument           DidDocument           `json:"didDocument"`
+	DidDocumentMetadata   DidDocumentMetadata   `json:"didDocumentMetadata"`
+	DidResolutionMetadata DidResolutionMetadata `json:"didResolutionMetadata"`
+}
+
+// DidDocument type.
+type DidDocument struct {
+	Context            []string             `json:"@context"`
+	Authentication     []string             `json:"authentication"`
+	ID                 string               `json:"id"`
+	Service            []Service            `json:"service"`
+	VerificationMethod []VerificationMethod `json:"verificationMethod"`
+}
+
+// Service type.
+type Service struct {
+	ID              string `json:"id"`
+	ServiceEndpoint string `json:"serviceEndpoint"`
+	Type            string `json:"type"`
+}
+
+// VerificationMethod type.
+type VerificationMethod struct {
+	Controller         string `json:"controller"`
+	ID                 string `json:"id"`
+	PublicKeyMultibase string `json:"publicKeyMultibase"`
+	Type               string `json:"type"`
+}
+
+// DidDocumentMetadata type.
+type DidDocumentMetadata struct {
+	Created   string `json:"created"`
+	VersionID string `json:"versionId"`
+}
+
+// DidResolutionMetadata type.
+type DidResolutionMetadata struct {
+	ContentType string `json:"contentType"`
+	Retrieved   string `json:"retrieved"`
+	DID         DID    `json:"did"`
+}
+
+// DID type.
+type DID struct {
+	DIDString        string `json:"didString"`
+	MethodSpecificID string `json:"methodSpecificId"`
+	Method           string `json:"method"`
+}
+
+func convertTypeRawDocCheqdToRawDoc(rawDocCheqd *RawDocCheqd) *rawDoc {
+	// Verification Method constants.
+	vmController := interface{}(rawDocCheqd.DidDocument.VerificationMethod[0].Controller)
+	vmID := interface{}(rawDocCheqd.DidDocument.VerificationMethod[0].ID)
+	vmPublicKeyMultibase := interface{}(rawDocCheqd.DidDocument.VerificationMethod[0].PublicKeyMultibase)
+	vmType := interface{}(rawDocCheqd.DidDocument.VerificationMethod[0].Type)
+
+	// Context constants.
+	context := make([]interface{}, len(rawDocCheqd.DidDocument.Context))
+	for i := 0; i < len(rawDocCheqd.DidDocument.Authentication); i++ {
```

Comment:
Should it be `len(rawDocCheqd.DidDocument.Context)`?

Reply:
Yep, It's correct

Reply:
Done! ✅